pkg/provider/aws/secretsmanager/secretsmanager.go

```diff
@@ -460,7 +460,7 @@ func (sm *SecretsManager) Validate() (esv1beta1.ValidationResult, error) {
 	}
 	_, err := sm.sess.Config.Credentials.Get()
 	if err != nil {
-		return esv1beta1.ValidationResultError, err
+		return esv1beta1.ValidationResultError, fmt.Errorf("Error retrieving AWS credentials")
```

Comment:
The `err` contains valuable information such as `IncompleteSignature`, i think it would be great if we can keep that information and scrub only the credentials.

We have this func:
https://github.com/external-secrets/external-secrets/blob/51e309258f1637159368459885a0c9783f90783c/pkg/provider/aws/util/errors.go#L27C12-L29

that removes the request-id from the error. Maybe we can extend it to scrub the credentials from the error.

Reply:
Hi @moolen, I've updated the PR with the proposed change. The regex is removing everything after the credentials because they can be pretty much anything and we are not going to lose much anyway.

Reply:
This sounds reasonable, thank you :heart: